src/System.Windows.Forms/src/System/Windows/Forms/DragDropHelper.cs

```diff
@@ -0,0 +1,369 @@
+            }
+            finally
+            {
+                if (dropTargetHelper is not null)
+                {
+                    Marshal.FinalReleaseComObject(dropTargetHelper);
+                }
+            }
+
+            return HRESULT.S_OK;
+        }
+
+        /// <summary>
+        ///  Notifies the drag-image manager that the cursor has left the drop target.
+        /// </summary>
+        public static HRESULT DragLeave()
+        {
+            if (TryGetDropTargetHelper(out IDropTargetHelper? dropTargetHelper) is false
+                || dropTargetHelper is null)
+            {
+                return HRESULT.S_FALSE;
+            }
+
+            try
+            {
+                HRESULT hr = dropTargetHelper.DragLeave();
+                if (!hr.Succeeded())
+                {
+                    Marshal.ThrowExceptionForHR((int)hr);
+                }
+            }
+            catch (COMException comEx)
+            {
+                Debug.WriteLineIf(CompModSwitches.DragDrop.TraceInfo, $"IDropTargetHelper::DragLeave COM error {comEx}");
+                return HRESULT.S_FALSE;
+            }
+            finally
+            {
+                if (dropTargetHelper is not null)
+                {
+                    Marshal.FinalReleaseComObject(dropTargetHelper);
+                }
+            }
+
+            return HRESULT.S_OK;
+        }
+
+        /// <summary>
+        ///  Notifies the drag-image manager that the object has been dropped, and provides it with the
+        ///  information needed to display the drag image.
+        /// </summary>
+        public static HRESULT Drop(IComDataObject dataObject, ref Point ppt, uint dwEffect)
+        {
+            if (TryGetDropTargetHelper(out IDropTargetHelper? dropTargetHelper) is false
+                || dropTargetHelper is null)
+            {
+                return HRESULT.S_FALSE;
+            }
+
+            try
+            {
+                HRESULT hr = dropTargetHelper.Drop(dataObject, ref ppt, dwEffect);
+                if (!hr.Succeeded())
+                {
+                    Marshal.ThrowExceptionForHR((int)hr);
+                }
+            }
+            catch (COMException comEx)
+            {
+                Debug.WriteLineIf(CompModSwitches.DragDrop.TraceInfo, $"IDropTargetHelper::Drop COM error {comEx}");
+                return HRESULT.S_FALSE;
+            }
+            finally
+            {
+                if (dropTargetHelper is not null)
+                {
+                    Marshal.FinalReleaseComObject(dropTargetHelper);
+                }
+            }
+
+            return HRESULT.S_OK;
+        }
+
+        /// <summary>
+        ///  Determines whether a drop description is present in a data object.
+        /// </summary>
+        public static bool GetDropDescriptionPresent(IComDataObject dataObject)
+        {
+            if (dataObject is null)
+                return false;
```

Comment:
Incorrect null annotations. Also don't skip on curlies.
```suggestion
        public static bool GetDropDescriptionPresent(IComDataObject? dataObject)
        {
            if (dataObject is null)
            {
                return false;
            }
```